File: backend/src/main/java/com/tripfriend/global/exception/ServiceException.java
package com.tripfriend.global.exception;

import com.tripfriend.global.dto.RsData;

public class ServiceException extends RuntimeException{

    private RsData<?> rsData;

    public ServiceException(String code, String msg){
        super(msg);
        rsData = new RsData<>(code, msg);
    }

    public String getCode(){
        return rsData.getCode();
    }

    public String getMsg(){
        return rsData.getMsg();
    }

    public int getStatusCode(){
        return rsData.getStatusCode();
    }

}

File: backend/src/main/java/com/tripfriend/global/exception/ServiceException.kt
package com.tripfriend.global.exception

import com.tripfriend.global.dto.Empty
import com.tripfriend.global.dto.RsData

class ServiceException(
    private val _code: String,
    override val message: String
) : RuntimeException(message) {
    private val rsData: RsData<Empty> = RsData(_code, message)

    val code: String
        get() = rsData.code

    val msg: String
        get() = rsData.msg

    val statusCode: Int
        get() = rsData.statusCode
}
